packages/Babylonian-Core.package/BPTraceValue.class/instance/belongingProbeLabel.st

```diff
@@ -0,0 +1,9 @@
+accessing
+belongingProbeLabel
+	
+	|annotations probe|
+	annotations := self containedMethodReference compiledMethod bpAnnotations.
+	probe := (annotations select: [:anObject | anObject isProbe]) 
+		ifNotEmpty: [:aCollection | aCollection detect: [:aProbe | aProbe id == self belongingProbeId]].
+	probe isCollection ifTrue: [self halt: 'Warning: received an empty collection, try adding a probe somewhere to fix it'].
```

Comment:
1) you use annotations only one time, so you don't need to save it in a variable
2) any `BPAnnotation` understands the message `id`. Therefore I would think that two annotations cannot have the same id. even if one is a probe and the other a slider, as they both are annotations. so i'd suggest calling detect directly on `annotations`
3) Collection has detect:ifFound:ifNotFound: (and other variations) so you would not need to check for if the collection is empty
4) i think a trace value should have a more generic function which just returns the belonging probe and accessing functions can call `label` on that

Reply:
My proposition:
```
belongingProbe

^ (self containedMethodReference compiledMethod bpAnnotations)
detect: [:anAnnotation | anAnnotation id == self belongingProbeId]
ifNone: [nil]

Reply:
after changed, only line 5 in packages/Babylonian-Printbugger.package/BPSwimlane.class/instance/newLineNameMorphFor..st must be changed and i think the tests should run through again